### PR TITLE
MAISTRA-2627 Remove namespace field from istiod-internal ClusterRoleBinding definition

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

**Review by dgn — APPROVED** [review]

**Review by jwendell — APPROVED** [review]

File: resources/helm/v2.0/istio-control/istio-discovery/templates/meshclusterrolebinding.yaml [hunk 1]
```diff
@@ -2,7 +2,6 @@ apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
   name: istiod-internal-{{ .Values.revision | default "default" }}-{{ .Release.Namespace }}
-  namespace: {{ .Release.Namespace }}
```
Comment (dgn) [inline resources/helm/v2.0/istio-control/istio-discovery/templates/meshclusterrolebinding.yaml:5]: This is going to be overwritten until the 2.0 PR goes in I guess

Comment (dgn) [reply resources/helm/v2.0/istio-control/istio-discovery/templates/meshclusterrolebinding.yaml:5]: Shoud we just not make the change here?

Comment (luksa) [reply resources/helm/v2.0/istio-control/istio-discovery/templates/meshclusterrolebinding.yaml:5]: That's why I'll wait for the 2.0 PR to be merged before merging this. If I remove this change and merge this 2.1 PR and then the 2.0 PR, we'll then get gen-check errors in all other 2.1 PRs until we run `make gen`. I thought just adding this change here was simpler.

**Pull request merged**